projects/cardiovascular-health/src/app/app.component.ts

```diff
@@ -10,16 +11,21 @@ import { DataLayerManagerService } from 'ngx-charts-on-fhir';
 export class AppComponent implements OnInit {
   showAddDataLayer: boolean = false;
   layers: any[] = [];
-  layerOrder: string[] = ['Heart rate', 'Blood Pressure', 'O2 Sat', 'Glucose', 'Step Count', 'Body Weight', 'Medications'];
+  isAllLayerSelected = true;
 
-  constructor(readonly layerManager: DataLayerManagerService) {}
+  constructor(readonly layerManager: DataLayerManagerService) { }
 
   ngOnInit(): void {
-    this.layerManager.retrieveAll(this.layerOrder);
+    this.layerManager.retrieveAll(this.orderLayers, this.isAllLayerSelected);
+  }
 
-    this.layerManager.availableLayers$.subscribe((layers) => {
-      layers.forEach((layer) => this.layerManager.select(layer.id));
-    });
+  orderLayers(layer: any) {
+    const layerOrder: string[] = ['Heart rate', 'Blood Pressure', 'O2 Sat', 'Glucose', 'Step Count', 'Body Weight', 'Medications']
+    return layer.pipe(
+      toArray(),
```

Comment:
`toArray` waits until the observable competes before emitting anything. I don't think that's what we want here